src/main/java/ch/securify/patterns/LockedEther.java

```diff
@@ -88,28 +100,31 @@ protected boolean isSafe(List<Instruction> instructions, AbstractDataflow datafl
 
     @Override
     protected boolean isViolation(List<Instruction> instructions, AbstractDataflow dataflow) {
-        boolean contractCanReceiveEther = false;
-
-        for (Instruction stopInstr : instructions) {
-            if (!(stopInstr instanceof Stop))
-                continue;
-
-            if (dataflow.instrMayDepOn(stopInstr, CallValue.class) == Status.UNSATISFIABLE) {
-                contractCanReceiveEther = true;
+        // check that the contract can receive ether
+        boolean contractCannotReceiveEther = true;
+        for (Instruction haltingInstr : instructions) {
+            if ((haltingInstr instanceof Stop || haltingInstr instanceof Return) && dataflow.instrMayDepOn(haltingInstr, CallValue.class) == Status.UNSATISFIABLE) {
```

Comment:
Why is this a necessary condition for the contract to receive ether?
Wouldn't a contract without a single CallValue instruction be able to receive Ether?

Reply:
In practice non-payable functions have a check on the value, and I think this was meant to look for this check (it was already there before). Shall I try to refine this?

Reply:
After reconsidering, I agree that it will probably be fine in practice.

I was thinking about cases like these, which are currently warnings, but are actually safe:

```js
contract Locked{
    function () payable external {
        msg.sender.transfer(100);
    }
}
```

But I think that they are very contrived.